db/view/row_locking.cc

```diff
@@ -170,8 +170,27 @@ row_locker::unlock(const dht::decorated_key* pk, bool partition_exclusive,
             lock.read_unlock();
         }
         if (!lock.locked()) {
+            auto& row_locks = pli->second._row_locks;
+            // We can erase the partition entry only if all its rows are unlocked.
+            // We don't expect any locked rows since the lock_holder is supposed
+            // to clean up any locked row by calling this function in its dtor.
+            // However, some unlocked rows may remain if lock_holder::lock_row
+            // failed to acquire the row after the row was created.
+            if (!row_locks.empty()) {
+                for (auto it = row_locks.begin(); it != row_locks.end(); ) {
+                    if (it->second.locked()) {
+                        mylog.warn("Encounered a locked row {} when unlocking partition {}", it->first, *pk);
```

Comment:
on_internal_error?

The whole thing seems like too much. Sure, we had a bug, but once we have proper nested locking we don't expect to encounter it again.

Reply:
I think this can be on_internal_error with only the serialized locking.
Will double check that.